force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,62 @@
+public with sharing class OpportunityTriggerHandler extends TriggerHandler {
+
+    // Private Collection Variables
+
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id, Opportunity> newOppMap;
+    private Map<Id, Opportunity> oldOppMap;
+
+    // Constructor
+
+    public OpportunityTriggerHandler() {
+        this.newOppList = (List<Opportunity>) Trigger.new;
+        this.oldOppList = (List<Opportunity>) Trigger.old;
+        this.newOppMap = (Map<Id,Opportunity>) Trigger.newMap;
+        this.oldOppMap = (Map<Id,Opportunity>) Trigger.oldMap;        
+    }
+
+    // Trigger Contexts
+
+    public override void beforeInsert() {
+        // Set default Type for new Opportunities
+        OpportunityHelper.setType(newOppList);
+    }
+
+    public override void beforeDelete() {
+        // Prevent deletion of closed Opportunities
+        OpportunityHelper.validateCloseOpportunity(oldOppList);
+
+        // Prevent deletion of a closed won opportunity if the account industry is 'Banking'.
+        OpportunityHelper.deleteCloseWonOpportunity(oldOppList);
+    }
+
+    public override void beforeUpdate() {
+        // Append Stage changes in Opportunity Description
+        // OpportunityHelper.updateDescriptionFromStage(Trigger.new, Trigger.oldMap); // <-- This doesn't work        
```

Comment:
Trigger.new nad Trigger.oldMap are available in the trigger context and not the class. The casting to newOppList will work in the class.